packages/store/types/tests/state-operators.remove-item.lint.ts

```diff
@@ -120,54 +120,58 @@ describe('[TEST]: the removeItem State Operator', () => {
       objs: [{ name: '1' }, { name: '2' }, { name: '3' }]
     };
 
+    patch<Original>({ nums: removeItem(item => item === '2') })(original); // $ExpectType Original
+    patch<Original>({ nums: removeItem(item => item === {}) })(original); // $ExpectType Original
+    patch<Original>({ nums: removeItem(item => item === true) })(original); // $ExpectType Original
+    patch<Original>({ nums: removeItem(item => item === []) })(original); // $ExpectType Original
+
+    patch<Original>({ strs: removeItem(item => item === 2) })(original); // $ExpectType Original
+    patch<Original>({ strs: removeItem(item => item === []) })(original); // $ExpectType Original
+    patch<Original>({ strs: removeItem(item => item === {}) })(original); // $ExpectType Original
+    patch<Original>({ strs: removeItem(item => item === true) })(original); // $ExpectType Original
+
+    patch<Original>({ bools: removeItem(item => item === '2') })(original); // $ExpectType Original
+    patch<Original>({ bools: removeItem(item => item === []) })(original); // $ExpectType Original
+    patch<Original>({ bools: removeItem(item => item === 2) })(original); // $ExpectType Original
+    patch<Original>({ bools: removeItem(item => item === {}) })(original); // $ExpectType Original
+
+    patch<Original>({ arrs: removeItem(item => item === {}) })(original); // $ExpectType Original
+    patch<Original>({ arrs: removeItem(item => item === 2) })(original); // $ExpectType Original
+    patch<Original>({ arrs: removeItem(item => item === '2') })(original); // $ExpectType Original
+    patch<Original>({ arrs: removeItem(item => item === true) })(original); // $ExpectType Original
+
+    patch<Original>({ objs: removeItem(item => item === 2) })(original); // $ExpectType Original
+    patch<Original>({ objs: removeItem(item => item === '2') })(original); // $ExpectType Original
+    patch<Original>({ objs: removeItem(item => item === true) })(original); // $ExpectType Original
+    patch<Original>({ objs: removeItem(item => item === []) })(original); // $ExpectType Original
```

Comment:
None of these should be valid! That is why they were explicitly listed as expected errors below.
It seems that this PR has broadened the typings too much.

Reply:
I can try to fix it, but for now it’s not quite clear how to do it without breaking another logic

Reply:
Anyway, I suggest working in this pull request, since we can track that the build is successful with Angular 8

Reply:
I disagree, we need to move forward in small steps. This is why I am breaking down my changes into small PRs.